Replace %USER% in lock history messages

MarcusP-P opened the pull request:
This replaces the %USER% with either the appropriate user's name, or the extension name in lock history.

I'm not sure if this is the best way to do this.

MarcusP-P committed:
Replace %USER% in lock history mesages

KittenApps commented on app/components/LockHistory.jsx:
```diff
@@ -9,12 +9,21 @@ import GetLockHistory from '../graphql/GetLockHistoryQuery.graphql';
       <Accordion>
         <AccordionSummary expandIcon={<ExpandMore/>}>
-          <Typography flexGrow={1}>{e.title}</Typography>
+          <Typography flexGrow={1}>{e.title.replace('%USER%', historyUserName(e))}</Typography>
```
```suggestion
          <Typography flexGrow={1}>{e.title.replace('%USER%', e.role === 'extension' ? e.extension : e.user?.username || 'unknown')}</Typography>
```

KittenApps reviewed (changes requested):
That's not the correct way to get the displayName of an extension. Showing the slug should do for now, until we get a proper extensions api in maybe 10 years hopefully.

MarcusP-P committed:
Update app/components/LockHistory.jsx

Fix this the correct way.

Co-authored-by: Silizia <66546362+KittenApps@users.noreply.github.com>

MarcusP-P committed:
Remove unneeded function

MarcusP-P commented:
I've added your patch, and removed the function. 

I assume the correct way of doing it would be to get the descriptions and slugs from the extensions endpoint and then look it up. That's a little beyond my React and GraphQL knowledge :)

chrmoritz commented:
and the current API for that is total garbage, because it doesn't scale either server nor client side with all these custom extension getting registered. So better wait until a proper solutions is developed eventually.

Thank you again.
